tests/test_Image.py

```diff
@@ -641,3 +641,27 @@ def test_plot_with_crs(self, s2_sr_vatican_2020, vatican, image_regression):
             fig.savefig(image_byte, format="png")
             image_byte.seek(0)
             image_regression.check(image_byte.getvalue())
+
+
+class TestFromList:
+    """Test ``fromList`` method."""
+
+    def test_from_list(self):
+        """This test is meant to fail."""
```

Comment:
any reason why you use such a complicated example when you could just create 3 constant images ? I tried in the code editor and your method is effectively necessary for this: 
```python 
sequence = ee.List.sequence(1, 3)
images = sequence.map(lambda i: ee.Image(ee.Number(i)).rename(ee.Number(i).int().format()))
bandNames = images.map(lambda i: ee.String(ee.Image(i).bandNames().get(0)))
var image = ee.ImageCollection(images).toBands().rename(bandNames)
image.bandNames().getinfo() == ["1", "2", "3"]
```

https://code.earthengine.google.com/f8eba4b7c59fb7043b32db27e86d4025

It also make me realized that my proposal implementation requires an extra step to get the `bandNames` (could  be use as a passive way to check if the band number is > 1)

Reply:
I used that case 'cos it was the only example I could find where it was needed, but I see you found another one.